notes/stv/20200119-AXS-3-Spark-2.txt

```diff
@@ -0,0 +1,295 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2020, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
+# On an existing Zeppelin/Spark cluster with AXS (see: 20201231-AXS-with-Hive.txt), try replacing with the following AXS distribution:
+# https://github.com/stevenstetzler/axs/releases/download/v3.0.0-preview/axs-distribution.tgz
+
+
+
+
+
+
+# -------------------------------------------------------------------
+# Check that pyspark works before changing everything:
+# fedora@zeppelin
+
+
+pyspark
+Python 3.7.7 (default, Mar 13 2020, 21:39:43) 
+[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+21/01/18 16:11:22 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
+Setting default log level to "WARN".
+To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
+21/01/18 16:11:23 WARN spark.SparkConf: Note that spark.local.dir will be overridden by the value set by the cluster manager (via SPARK_LOCAL_DIRS in mesos/standalone/kubernetes and LOCAL_DIRS in YARN).
+21/01/18 16:11:24 WARN yarn.Client: Neither spark.yarn.jars nor spark.yarn.archive is set, falling back to uploading libraries under SPARK_HOME.
+Welcome to
+      ____              __
+     / __/__  ___ _____/ /__
+    _\ \/ _ \/ _ `/ __/  '_/
+   /__ / .__/\_,_/_/ /_/\_\   version 2.4.3
+      /_/
+
+Using Python version 3.7.7 (default, Mar 13 2020 21:39:43)
+SparkSession available as 'spark'.
+
+
+
+# -------------------------------------------------------------------
+# Try replacing the Spark installation in our Zeppelin node:
+# fedora@zeppelin
+
+sudo su
+pushd /opt	
+    mv spark-3.0.0-preview-bin-AXS-v3.0.0-preview spark-2.4.7
+    wget https://github.com/stevenstetzler/axs/releases/download/v3.0.0-preview/axs-distribution.tgz
+    tar -xzvf axs-distribution.tar.gz
+    rm -f axs-distribution.tar.gz 
```

Comment:
This appears to rename the existing AXS 3.0.0 version from `spark-3.0.0-preview-bin-AXS-v3.0.0-preview` to `spark-2.4.7`.

It then downloads version 3.0.0-preview of the AXS distribution from a [3rd party GitHub account,](https://github.com/stevenstetzler/axs) and unpacks it as `spark-3.0.0-preview-bin-AXS-v3.0.0-preview`, replacing the original copy of `spark-3.0.0-preview-bin-AXS-v3.0.0-preview`.

So what we end up with in the `/opt` directory is :

- `spark-2.4.7` .... (originally `spark-3.0.0-preview-bin-AXS-v3.0.0-preview`)
- `spark-3.0.0-preview-bin-AXS-v3.0.0-preview` .... (new download from stevenstetzler)

What is missing from the notes is that /opt contains a symlink `/opt/spark` that selects the current active version of Spark.

Better to have left the existing AXS 3.0.0 version where it was, download the version from stevenstetzler and unpack it as `spark-3.0.0-preview-bin-AXS-v3.0.0-stevenstetzler`, and then use the symlink to select the downloaded version.

- `spark-3.0.0-preview-bin-AXS-v3.0.0-preview` .... (as it was)
- `spark-3.0.0-preview-bin-AXS-v3.0.0-stevenstetzler` .. (new download)
- `spark -> spark-3.0.0-preview-bin-AXS-v3.0.0-stevenstetzler` ... (symlink)